src/domain/FlowRunner.ts

```diff
@@ -578,7 +578,7 @@ export class FlowRunner implements IFlowRunner, IFlowNavigator, IPromptBuilder {
     ctx.cursor = this.dehydrateCursor(richCursor)
 
     // todo: this could be findFirstExitOnActiveFlowBlockFor to an Expressions Behaviour
-    this.cacheInteractionByBlockName(richCursor.interaction, block as IMessageBlock, this.context)
+    this.cacheInteractionByBlockName(richCursor.interaction, block as IMessageBlock, ctx)
```

Comment:
@AlteaDown — This was just a noticing from our last debugging session where we noticed this was an oddball instance of not referencing provided context and rather pulling from `this` instead.

Reply:
I find that usually, when a function has a parameter of some type, but uses a the reference already in the class field, it means the parameter is redundant. Is that the case here?

Reply:
I understand this perspective, you're probably right.

My internal struggle is the balance between injecting data so that unit tests are dirt simple vs. having to preconfigure an instance before invoking. So, for methods with primarily internal use, I often "unnecessarily" provide `thing` so that the tests can be `subject.doActionWith(thing)`.

Thoughts?

Reply:
I thought that's likely the case here. Not necessary, but something to consider. It depends how the dependency is injected into the field, (because sometimes it's just as easy to mock the dependency before injecting). It would be nice to have some sort of DI tooling setup to help with this in the future as well.